tests/components/mold_indicator/test_sensor.py

```diff
@@ -12,284 +12,280 @@
-        )
-        self.hass.states.set(
-            "test.indoorhumidity",
-            STATE_UNKNOWN,
-            {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE},
-        )
-        self.hass.block_till_done()
-        moldind = self.hass.states.get("sensor.mold_indicator")
-        assert moldind
-        assert moldind.state == "unavailable"
-        assert moldind.attributes.get(ATTR_DEWPOINT) is None
-        assert moldind.attributes.get(ATTR_CRITICAL_TEMP) is None
-
-        self.hass.states.set(
-            "test.indoorhumidity", "20", {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE}
-        )
-        self.hass.block_till_done()
-        moldind = self.hass.states.get("sensor.mold_indicator")
-        assert moldind
-        assert moldind.state == "23"
-
-        dewpoint = moldind.attributes.get(ATTR_DEWPOINT)
-        assert dewpoint
-        assert dewpoint > 4.58
-        assert dewpoint < 4.59
-
-        esttemp = moldind.attributes.get(ATTR_CRITICAL_TEMP)
-        assert esttemp
-        assert esttemp == 27.5
-
-    def test_sensor_changed(self):
-        """Test the sensor_changed function."""
-        assert setup_component(
-            self.hass,
-            sensor.DOMAIN,
-            {
-                "sensor": {
-                    "platform": "mold_indicator",
-                    "indoor_temp_sensor": "test.indoortemp",
-                    "outdoor_temp_sensor": "test.outdoortemp",
-                    "indoor_humidity_sensor": "test.indoorhumidity",
-                    "calibration_factor": 2.0,
-                }
-            },
-        )
-        self.hass.block_till_done()
-        self.hass.start()
-
-        self.hass.states.set(
-            "test.indoortemp", "30", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
-        )
-        self.hass.block_till_done()
-        assert self.hass.states.get("sensor.mold_indicator").state == "90"
-
-        self.hass.states.set(
-            "test.outdoortemp", "25", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
-        )
-        self.hass.block_till_done()
-        assert self.hass.states.get("sensor.mold_indicator").state == "57"
-
-        self.hass.states.set(
-            "test.indoorhumidity", "20", {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE}
-        )
-        self.hass.block_till_done()
-        assert self.hass.states.get("sensor.mold_indicator").state == "23"
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture(autouse=True)
+def init_sensors_fixture(hass):
+    """Set up things to be run when tests are started."""
+    hass.states.async_set(
+        "test.indoortemp", "20", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    hass.states.async_set(
+        "test.outdoortemp", "10", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    hass.states.async_set(
+        "test.indoorhumidity", "50", {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE}
+    )
+
+
+async def test_setup(hass):
+    """Test the mold indicator sensor setup."""
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)